testing/run_tests.py

```diff
@@ -58,12 +65,13 @@ def RunCmd(cmd, forbidden_output=[], expect_failure=False, env=None, **kwargs):
   stdout, stderr = process.communicate()
   end_time = time.time()
 
-  if process.returncode != 0 and not expect_failure:
+  if (expect_failure and process.returncode == 0) or (not expect_failure and
```

Comment:
It looks like `expect_failure` was being used more as "failure is okay":

https://github.com/flutter/engine/pull/34245/files#diff-521d7c59c3f721b259f094c760c2613e16e889b0f24702280924466109e3b0b0R677

Reply:
I think if Jason's concerns are addressed, then any way that you can find to keep the pre-existing tests working in line with their intent is fine with me. That could mean splitting this out into two flags, e.g. "ignore_failure" and "expect_failure".